### PR TITLE
[Android] Update submit validation API

### PR DESCRIPTION
**Pull request by almedina-ms**
## Related Issue
Relates to #4829 

## Description
Updates the SubmitValidation API removing the use of the shared model `InternalId`, the usage of this properties led to app crashes due to the retrieval of the values taking too long. 

The update also includes the `registerSubmitableAction` which lets the action listeners to know what actions should validation be performed for, this helps avoid unnecessary validations on custom elements that may not need it. 

It also adds an `areInputsValid` which lets the developer know what the result of the last performed validation was, this removes the need to re-execute the validation process to know the validation result.

## How Verified
The fix was verified by adding a sample CustomBlueAction and CustomActionListener which makes use of the previously described APIs

CustomBlueAction

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/4876)

### CODE REVIEW

**Comment by shalinijoshi19** [comment]
> ## Related Issue
> Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
> the issues fixed with this PR (eg Fixes #). If an issue doesn't yet exist please create one if reasonable to aid
> in issue tracking.
> 
> **NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
> page above.
> 
> ## Description
> Create a [descriptive title and provide all of the relevant details in the description](https://chris.beams.io/posts/git-commit/).
> This information lives with the code and will help future readers (_**including yourself**_) and will serve as documentation.
> 
> At the very least please describe the issue you are addressing, and what your fix entails.
> 
> ## How Verified
> How you verified the fix, including one or all of the following:
> 
> 1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
> 2. Existing relevant unit/regression tests that you ran
> 3. Manual scenario verification if any; _**Do include .gif's or screenshots of the testing you performed here if you think that it
>    will aid in code reviews or corresponding fixes on other platforms for eg.**_
> 
> ###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/4876)

@almedina-ms can you remove the placeholder PR description text from the template and update with relevant description/details around this PR? thanks

**Review by shalinijoshi19 — CHANGES_REQUESTED** [review]
Could you start by updating the description? thanks

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ContainerRenderer.java [hunk 1]
```diff
@@ -224,11 +226,16 @@ public static void setBackgroundImage(RenderedAdaptiveCard renderedCard,
         }
     }
 
-    public static void setSelectAction(RenderedAdaptiveCard renderedCard, BaseActionElement selectAction, View view, ICardActionHandler cardActionHandler)
+    public static void setSelectAction(RenderedAdaptiveCard renderedCard, BaseActionElement selectAction, View view, ICardActionHandler cardActionHandler, RenderArgs renderArgs)
```
Comment (shalinijoshi19) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ContainerRenderer.java:229]: Can we add a unit test for the public API so we have automated test coverage for this scenario going forward?

**Review by shalinijoshi19 — CHANGES_REQUESTED** [review]
Can you include a unit test for the public API to cover the crash scenarios to ensure automated coverage going forward? Thanks

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ContainerRenderer.java (hunk 1 above)
Comment (almedina-ms) [reply source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ContainerRenderer.java:224]: I added a unit test that covers the API

File: source/android/adaptivecards/src/androidTest/java/io/adaptivecards/objectmodel/APITest.java [hunk 2]
```diff
@@ -1,15 +1,32 @@
 package io.adaptivecards.objectmodel;
 
 import android.content.Context;
+import android.os.AsyncTask;
 import android.support.test.InstrumentationRegistry;
+import android.support.v4.app.FragmentActivity;
+import android.support.v4.app.FragmentManager;
```
Comment (shalinijoshi19) [inline source/android/adaptivecards/src/androidTest/java/io/adaptivecards/objectmodel/APITest.java:7]: if you are on here @almedina-ms can you add in the msft copyright header pls

**Review by shalinijoshi19 — APPROVED** [review]
:shipit:

**Pull request merged**